yamux/src/connection.rs

```diff
@@ -409,6 +401,14 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Active<T> {
                     self.socket.start_send_unpin(frame)?;
                     continue;
                 }
+
```

Comment:
Changing this order seem to fix the issue.

Reply:
Does it deadlock on yamux test too? If not, maybe we could add a test for this too?

Reply:
Could you please elaborate? I understand this test was designed specifically to address this type of deadlock.

Reply:
I actually missed the test that was there so my comment can be ignored :)